server/src/main/java/io/crate/execution/ddl/tables/AlterTableRequest.java

```diff
@@ -25,69 +25,94 @@
 import static org.elasticsearch.common.settings.Settings.writeSettingsToStream;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 
 import org.elasticsearch.Version;
 import org.elasticsearch.action.support.master.AcknowledgedRequest;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.settings.Settings;
-import org.jetbrains.annotations.Nullable;
 
+import io.crate.metadata.PartitionName;
 import io.crate.metadata.RelationName;
 
 public class AlterTableRequest extends AcknowledgedRequest<AlterTableRequest> {
 
-    private final RelationName tableIdent;
-    @Nullable
-    private final String partitionIndexName;
+    private final RelationName table;
+    private final List<String> partitionValues;
     private final boolean isPartitioned;
```

Comment:
strictly speaking `isPartitioned` is also redundant - the master can easily tell that a table is partitioned.
But can't set the right default for bwc streaming, so until we drop support for nodes that made use of the value in the master action we can't remove it.